src/visualizations/config/adapters/dhis_highcharts/series/index.js

```diff
@@ -168,16 +173,29 @@ function getDefault(series, metaData, layout, isStacked, extraOptions) {
             seriesObj.groupPadding = 0
         }
 
-        const legendSet = extraOptions?.legendSets?.find(
-            legendSet =>
-                legendSet.id === metaData.items[seriesObj.id]?.legendSet
-        )
+        let legendSet
+        if (isLegendSetType(layout.type)) {
+            const legendSets = extraOptions?.legendSets || []
+            if (
+                layout.legendDisplayStrategy ===
+                LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM
+            ) {
+                legendSet = legendSets.find(
+                    legendSet =>
+                        legendSet.id === metaData.items[seriesObj.id]?.legendSet
+                )
+            } else if (
+                layout.legendDisplayStrategy === LEGEND_DISPLAY_STRATEGY_FIXED
+            ) {
+                legendSet = legendSets[0]
```

Comment:
I assume in this case only the selected one in options is in the `legendSets` list?

Reply:
Yeah this follows the pattern introduced here: https://github.com/dhis2/analytics/blob/402cb78c3e4b8c3dd48ed4e0001854a30af2b4a0/src/visualizations/config/adapters/dhis_highcharts/index.js#L201